### ISSUE TITLE
parenthesis and other token matching plugin that supports treesitter written in Lua

### ISSUE THREAD

**Issue by comiluv** [nvim-lua/wishlist#48]
**What?**
Press % to jump between matching parens, if ... else ... end blocks, function() ... end blocks, highlight those keywords, etc.
Basicially github.com/andymass/vim-matchup

**Why?**
As the Neovim's default plugins matchit and matchparens are not going to support treesitter anytime soon (not a fact but a speculation), a lua version of something that does this would be nice

**Potential existing implementations:**
github.com/andymass/vim-matchup

**Potential pitfalls*:*
afaik, treesitter integration to Neovim is dependent to nvim-treesitter and its plugins, so any plugin that works on nvim-treesitter would be prone to bugs and errors in those plugins.

**Comment by Lazerbeak12345** [comment]
I'm not sure _how_ I did it, but it works in my config. https://github.com/Lazerbeak12345/Lazerbeak12345

**Comment by comiluv** [comment]
Yes "matchit" is a plugin written in viml and is by default included in official distribution of Neovim.
But a version similar to matchit that is written in lua doesn't exist (AFAIK) and this wishlist item was written because of a lua version of same functionality can be desirable.
